rust_src/src/buffers.rs

```diff
@@ -1402,6 +1403,22 @@ fn get_truename_buffer_1(filename: LispSymbolOrString) -> LispObject {
         .into()
 }
 
+/// Recenter the overlays of the current buffer around position POS.
+/// That makes overlay lookup faster for positions near POS (but perhaps slower
+/// for positions far away from POS).
+#[lisp_fn]
+pub fn overlay_recenter(pos: LispObject) -> LispObject {
```

Comment:
I am somewhat certain you can use `LispNumber` here, it uses coerce internally. Then below you use `pos.into()` to get the EmacsInt aka fixnum. Since this only returns Qnil you do not need to specify a return.

```suggestion
pub fn overlay_recenter(pos: LispNumber) {
```

Reply:
I had to use `to_fixnum()` on the call to `clip_to_bounds`, because the compiler complained that the `From<numbers::LispNumber>` trait wasn't implemented for i64. Is that the correct way?